### PR TITLE
[Core worker] Task execution passes TaskInfo struct to executor

### PR DESCRIPTION
**Pull request by kfstorm**
## What do these changes do?
I'm currently working on Java worker refactor based on core worker. Some necessary fields are required for Java worker to execute tasks.

- Driver ID is required for Java worker to get `RayFunction` from `FunctionManager`.
https://github.com/ray-project/ray/blob/1a39fee9c6fa0f81e1184182b47ef106269613fe/java/runtime/src/main/java/org/ray/runtime/Worker.java#L89

- Task type (`isActorTask`, `isActorCreationTask`) is required for exception handling and actor checkpoint. Although task type can be generated from `RayFunction`, but when failed to get `RayFunction` from `FunctionManager`, it's unable to handle exception for actor creation task. 
https://github.com/ray-project/ray/blob/1a39fee9c6fa0f81e1184182b47ef106269613fe/java/runtime/src/main/java/org/ray/runtime/Worker.java#L100
https://github.com/ray-project/ray/blob/1a39fee9c6fa0f81e1184182b47ef106269613fe/java/runtime/src/main/java/org/ray/runtime/Worker.java#L100
https://github.com/ray-project/ray/blob/1a39fee9c6fa0f81e1184182b47ef106269613fe/java/runtime/src/main/java/org/ray/runtime/Worker.java#L129

## Related issue number

#4850

## Linter

- [x] I've run `scripts/format.sh` to lint the changes in this PR.

### CODE REVIEW

**Comment by zhijunfu** [comment]
LGTM, thanks.

Could you add a bit more details in the description on why these new fields are required? 

btw for better tracking, you can also link this pr to the core worker issue, or create a new issue for migrating java worker and link it to that one.

File: src/ray/core_worker/common.h [hunk 1]
```diff
@@ -68,6 +68,18 @@ class TaskArg {
...
+/// Information of a task
+struct TaskInfo {
+  /// The ID of task.
+  const TaskID task_id;
+  /// The driver ID.
+  const DriverID driver_id;
+  /// Whether this task is to create an actor.
+  const bool is_actor_creation_task;
+  /// Whether this task is to execute a function of an actor.
+  const bool is_actor_task;
```
Comment (jovany-wang) [inline src/ray/core_worker/common.h]: Use an enum `Type` to indicate the type of this task?

**Review by jovany-wang — APPROVED** [review]
LGTM except a minor comment!

File: src/ray/core_worker/common.h (hunk 1 above)
Comment (kfstorm) [reply src/ray/core_worker/common.h]: Great idea!

**Comment by kfstorm** [comment]
@zhijunfu Description added.

**Comment by raulchen** [comment]
Should we just pass in the whole task spec? In case we need more info in the future.

**Comment by kfstorm** [comment]
> Should we just pass in the whole task spec? In case we need more info in the future.

@zhijunfu told me that he hided some fields on purpose that he doesn't want to expose all fields.

**Comment by jovany-wang** [comment]
Could you rebase the master to fix the linting issue?

**Pull request merged**